### PR TITLE
JBPM-8347 Increate Prometheus metrics test coverage

### PR DESCRIPTION
**Pull request by sutaakar**
Also contains reproducer for JBPM-8452

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java [hunk 1]
```diff
@@ -136,4 +157,76 @@ public void testPrometheusProcessAndTaskMetrics() {
...
+        assertThat(getMetrics()).contains(
+              "kie_server_case_started_total{case_definition_id=\"" + CASE_DEF_ID + "\",",
+              "kie_server_case_running_total{case_definition_id=\"" + CASE_DEF_ID + "\",} 1.0"
+        );
+
+        caseClient.cancelCaseInstance(CONTAINER_ID_CASE, caseId);
+
+        assertThat(getMetrics()).contains(
+              "kie_server_case_started_total{case_definition_id=\"" + CASE_DEF_ID + "\",",
+              "kie_server_case_running_total{case_definition_id=\"" + CASE_DEF_ID + "\",} 0.0",
```
Comment (vblagoje) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: @sutaakar Any potential timing issues here to get the counter to be 0?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java [hunk 2]
```diff
@@ -136,4 +157,76 @@ public void testPrometheusProcessAndTaskMetrics() {
...
+        Long jobIdTomorrow = jobServicesClient.scheduleRequest(jobRequestInstanceTomorrow);
+        jobServicesClient.cancelRequest(jobIdTomorrow);
+
+        JobRequestInstance jobRequestInstanceNow = createJobRequestInstance();
+        Long jobIdNow = jobServicesClient.scheduleRequest(jobRequestInstanceNow);
+        KieServerSynchronization.waitForJobToFinish(jobServicesClient, jobIdNow);
+
+        assertThat(getMetrics()).contains(
+              "kie_server_job_scheduled_total{container_id=\"\",command_name=\"" + PRINT_OUT_COMMAND + "\",}",
+              "kie_server_job_cancelled_total{container_id=\"\",command_name=\"" + PRINT_OUT_COMMAND + "\",} " + (currentNumberOfCancelled + 1)
```
Comment (vblagoje) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: @sutaakar here you clearly wait for the jobs to finish...so should be ok.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java [hunk 3]
```diff
@@ -136,4 +157,76 @@ public void testPrometheusProcessAndTaskMetrics() {
...
+              "kie_server_case_duration_seconds_sum{case_definition_id=\"" + CASE_DEF_ID + "\","
+        );
+    }
+
+    @Test
+    public void testPrometheusJobMetrics() throws Exception {
+        int currentNumberOfCancelled = jobServicesClient.getRequestsByStatus(Collections.singletonList(STATUS.CANCELLED.toString()), 0, 1000).size();
+        int currentNumberOfDone = jobServicesClient.getRequestsByStatus(Collections.singletonList(STATUS.DONE.toString()), 0, 1000).size();
+
+        Calendar tomorrow = Calendar.getInstance();
```
Comment (vblagoje) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: Nitpicking here. Maybe a one-liner will be more clear?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java (hunk 1 above)
Comment (sutaakar) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: @vblagoje I am not sure if I get your question. The case is cancelled above so the number of running cases should decrement to 0.0. The decrement should happen in listener while cancelCaseInstance() is processed. There should be no timing issues.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java (hunk 3 above)
Comment (sutaakar) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: Good  point, adjusted.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java (hunk 1 above)
Comment (vblagoje) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: Ok cool. Just checking

Comment (sutaakar) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/prometheus/PrometheusIntegrationTest.java]: In the end  had to delete the check for counter value. Problem is that other tests don't close/delete cases correctly when they finish, causing counter to become inconsistent with actual opened cases.

**Comment by sutaakar** [comment]
@mswiderski Can you please take a look?

**Pull request merged**